cpp/test/sg/fil_test.cu

```diff
@@ -828,6 +893,12 @@ std::vector<FilTestParams> predict_sparse_inputs = {
                   threads_per_tree = 64),
   FIL_TEST_PARAMS(num_rows = 500, num_cols = 2000, algo = ALGO_AUTO,
                   threads_per_tree = 256, n_items = 1),
+  FIL_TEST_PARAMS(num_trees = 51, leaf_algo = VECTOR_LEAF, num_classes = 15),
+  FIL_TEST_PARAMS(leaf_algo = VECTOR_LEAF, num_trees = 9, num_classes = 20),
+  FIL_TEST_PARAMS(num_rows = 103, num_cols = 1000, depth = 5, num_trees = 1,
```

Comment:
Could you also add a test with `num_trees > FIL_TPB`, different class counts (e.g., 11 and 1111) and a small number of rows (103 would do), to test the case where accumulation is performed several times per thread block?